core_backend/app/utils.py

```diff
@@ -6,6 +7,11 @@
 from .config import LOG_LEVEL
 
 
+def get_key_hash(retrieval_key: str) -> str:
+    """Hashes the retrieval key using SHA256."""
+    return hashlib.sha256(retrieval_key.encode()).hexdigest()
```

Comment:
Do you think this is enough security for now or should we go with a stronger algorithm?

Reply:
Yep! I agree. Thank you.